gobblin-runtime/src/main/java/org/apache/gobblin/runtime/GobblinMultiTaskAttempt.java

```diff
@@ -400,15 +403,20 @@ private boolean taskSuccessfulInPriorAttempt(String taskId) {
           // task could not be created, so directly count down
           countDownLatch.countDown();
           log.error("Could not create task for workunit {}", workUnit, e);
-        } else if (!task.hasTaskFuture()) {
-          // Task was created and may have been registered, but not submitted, so call the
-          // task state tracker task run completion directly since the task cancel does nothing if not submitted
-          this.taskStateTracker.onTaskRunCompletion(task);
-          log.error("Could not submit task for workunit {}", workUnit, e);
         } else {
-          // task was created and submitted, but failed later, so cancel the task to decrement the CountDownLatch
-          task.cancel();
-          log.error("Failure after task submitted for workunit {}", workUnit, e);
+          if (!task.hasTaskFuture()) {
+            // Task was created and may have been registered, but not submitted, so call the
+            // task state tracker task run completion directly since the task cancel does nothing if not submitted
+            this.taskStateTracker.onTaskRunCompletion(task);
+            log.error("Could not submit task for workunit {}", workUnit, e);
+          } else {
+            // task was created and submitted, but failed later, so cancel the task to decrement the CountDownLatch
+            task.cancel();
+            log.error("Failure after task submitted for workunit {}", workUnit, e);
+          }
+
+          // Remove entry in the metrics registry when task failed to be executed .
+          this.cleanMetrics();
```

Comment:
Should we be cleaning metrics when task execution fails? Currently, cleanMetrics() will clean up metrics for all tasks associated with a Multi-task attempt. Won't we delete metrics for tasks which may be currently running?

Reply:
Looks like I shouldn't clean the metric registry since it is a class-static variable shared by all instances. Adjust the logic.